proto/zitadel/system.proto

```diff
@@ -0,0 +1,611 @@
+        option (google.api.http) = {
+            delete: "/failedevents/{database}/{view_name}/{failed_sequence}";
+        };
+
+        option (zitadel.v1.auth_option) = {
+            permission: "iam.write";
+        };
+
+        option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+            tags: "failed events";
+            external_docs: {
+                url: "https://docs.zitadel.ch/concepts#Software_Architecture";
+                description: "details of ZITADEL's event driven software concepts";
+            };
+            responses: {
+                key: "200";
+                value: {
+                    description: "Events removed from the list";
+                };
+            };
+            responses: {
+                key: "400";
+                value: {
+                    description: "failed event not found";
+                    schema: {
+                        json_schema: {
+                            ref: "#/definitions/rpcStatus";
+                        };
+                    };
+                };
+            };
+        };
+    }
+}
+
+
+//This is an empty request
+message HealthzRequest {}
+
+//This is an empty response
+message HealthzResponse {}
+
+message ListInstancesRequest {
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_schema) = {
+		json_schema: {
+			description: "Search query for lists";
+			required: ["query"]
+		};
+	};
+
+    //list limitations and ordering
+    zitadel.v1.ListQuery query = 1;
+    // the field the result is sorted
+    zitadel.instance.v1.InstanceFieldName sorting_column = 2;
+    //criterias the client is looking for
+    repeated zitadel.instance.v1.OrgQuery queries = 3;
+}
+
+message ListInstancesResponse {
+    zitadel.v1.ListDetails details = 1;
+    zitadel.instance.v1.OrgFieldName sorting_column = 2;
+    repeated zitadel.instance.v1.Instance result = 3;
+}
+
+message GetInstanceRequest {
+    string domain = 1 [(validate.rules).string = {min_len: 1, max_len: 200}];
+}
+
+message GetInstanceResponse {
+    zitadel.instance.v1.Instance instance = 1;
+}
+
+message AddInstanceRequest {
+    string domain = 1 [(validate.rules).string = {min_len: 1, max_len: 200}];
+    string instance_name = 2 [(validate.rules).string = {min_len: 1, max_len: 200}];
+    string first_org_name = 3 [(validate.rules).string = {min_len: 1, max_len: 200}];
+    zitadel.instance.v1.Region region = 4;
+    string owner_first_name = 5 [(validate.rules).string = {min_len: 1, max_len: 200}];
+    string owner_last_name = 6 [(validate.rules).string = {min_len: 1, max_len: 200}];
+    string owner_email = 7 [(validate.rules).string = {min_len: 1, max_len: 200}];
+    string owner_username = 8 [(validate.rules).string = {min_len: 1, max_len: 200}];
+    string password = 9 [(validate.rules).string = {min_len: 1, max_len: 200}];
+    string subscription_name = 10;
+    uint64 request_limit = 11;
+    uint64 action_mins_limit = 12;
```

Comment:
hard cap ?
and where do we define the following "limits" / "portions"? e.g. if the instance has no hard cap and has exceeded the request_limit and now automatically gets unlocked for another amount of requests